dpctl-capi/source/dpctl_sycl_device_manager.cpp

```diff
@@ -146,15 +146,18 @@ DPCTLDeviceMgr_GetDevices(int device_identifier)
     } catch (std::bad_alloc const &ba) {
         return nullptr;
     }
-    auto &cache = DeviceCacheBuilder::getDeviceCache();
+    const auto root_devices = device::get_devices();
```

Comment:
You need `const auto& root_devices = device::get_devices();` without the `&` a copy of the vector will be performed.